packages/material-ui/src/Badge/Badge.js

```diff
@@ -200,59 +236,42 @@ const Badge = React.forwardRef(function Badge(props, ref) {
     invisible = true;
   }
 
-  const {
-    anchorOrigin = anchorOriginProp,
-    badgeContent,
-    color = colorProp,
-    max = maxProp,
-    overlap = overlapProp,
-    variant = variantProp,
-  } = invisible ? prevProps : props;
+  const { color = colorProp } = invisible ? prevProps : props;
 
-  const themeVariantsClasses = useThemeVariants(
-    {
-      ...props,
-      anchorOrigin,
-      badgeContent,
-      color,
-      component: ComponentProp,
-      invisible,
-      max,
-      overlap,
-      showZero,
-      variant,
-    },
-    'MuiBadge',
-  );
-
-  let displayValue = '';
-
-  if (variant !== 'dot') {
-    displayValue = badgeContent > max ? `${max}+` : badgeContent;
-  }
+  const classes = extendBadgeClasses({ ...props, invisible, color });
 
   return (
-    <ComponentProp className={clsx(classes.root, className)} ref={ref} {...other}>
-      {children}
-      <span
-        className={clsx(
-          classes.badge,
-          classes[variant],
-          classes[
-            `anchorOrigin${capitalize(anchorOrigin.vertical)}${capitalize(
-              anchorOrigin.horizontal,
-            )}${capitalize(overlap)}`
-          ],
-          {
-            [classes[`color${capitalize(color)}`]]: color !== 'default',
-            [classes.invisible]: invisible,
-          },
-          themeVariantsClasses,
-        )}
-      >
-        {displayValue}
-      </span>
-    </ComponentProp>
+    <BadgeUnstyled
+      {...{
+        color: colorProp,
+        invisible: invisibleProp,
+        badgeContent: badgeContentProp,
+        showZero,
+        variant: variantProp,
+        ...other,
+      }}
```

Comment:
So we keep using JSX?

```suggestion
      color={colorProp}
      invisible={invisibleProp}
      badgeContent={badgeContentProp}
      showZero={showZero}
      variant={variantProp}
      {...other}
```